charts/cryostat/templates/auth_proxy.yaml

```diff
@@ -0,0 +1,19 @@
+version: "3"
+services:
```

Comment:
I don't think this will work... ? This is a docker-compose definition, but what we need in the helm chart are, well, helm resources, which are templated versions of Kubernetes resources.

Take a look at this file in my PR for example: https://github.com/cryostatio/cryostat-helm/pull/111/files#diff-45046576393413b4292b320376e53136d68615d83d721b3fde4970e5985098a5

Reply:
I think you should also rebase on top of #111 , otherwise you'll be trying to put an auth proxy in front of Cryostat 2.4, which is quite a different setup (no database, no storage container, etc.)